File: boat-sample/src/test/java/sample/java/xyz/srclab/core/invoke/InvokeSample.java
package sample.java.xyz.srclab.core.invoke;

import org.testng.annotations.Test;
import xyz.srclab.common.func.Invoker;
import xyz.srclab.common.test.TestLogger;

public class InvokeSample {

    private static final TestLogger logger = TestLogger.DEFAULT;

    @Test
    public void testInvoke() throws Exception {
        Invoker invoker = Invoker.forMethod(String.class.getMethod("getBytes"));
        byte[] bytes = invoker.invoke("10086");
        //[49, 48, 48, 56, 54]
        logger.log("bytes: {}", bytes);
    }
}

File: boat-sample/src/test/kotlin/sample/kotlin/xyz/srclab/core/invoke/InvokeSample.kt
package sample.kotlin.xyz.srclab.core.invoke

import org.testng.annotations.Test
import xyz.srclab.common.invoke.Invoker
import xyz.srclab.common.test.TestLogger

class InvokeSample {

    @Test
    fun testInvoke() {
        val invoker: Invoker = Invoker.forMethod(String::class.java, "getBytes")
        val bytes = invoker.invoke<ByteArray>("10086")
        //[49, 48, 48, 56, 54]
        logger.log("bytes: {}", bytes)
    }

    companion object {
        private val logger = TestLogger.DEFAULT
    }
}